code/modules/organs/external/_external.dm

```diff
@@ -325,6 +331,15 @@
 
 	return all_items
 
+/obj/item/organ/external/proc/get_limbs_recursive(var/no_stumps = FALSE)
+	var/list/all_limbs = list()
+	for(var/obj/item/organ/external/child in children)
+		if(no_stumps && child.is_stump())
+			continue
+		all_limbs += child
```

Comment:
> Are there assurances that children can't be cyclic? It might be good to check if it's already in the list, and only add it and call get_limbs_recursive if it isn't.

Childrens have to be unique in the mob's list and organs list. If there were cyclic references, this proc would be the last of your worries, cause everything would start breaking up. (True story XD )

> (Also, would kind of like to see either separate functions instead of arguments like `no_stumps`, or the ability to pass predicates/procs/callbacks in. A whitelist and blacklist argument would probably suffice, so to replicate `no_stumps` you'd pass a callback to `is_stump`, or the path to is_stump, as the blacklist argument. Kind of a shame there are no inline functions in DM, but I think it's a better design than control coupling.)

I guess that could be a future improvement. But its a bit out of scope right now. This is mainly a bugfix PR XD
Truth be told, a lot of the human code is still really wonky and will need work, so I'm not too much in a hurry to start polishing stuff like that, and having it get changed anyways a few weeks later when someone refactors all this.